tests/utils/rpc/request.go

```diff
@@ -93,6 +94,48 @@ func Decode(body []byte, target interface{}) error {
 	return nil
 }
 
+// DecodeScale decodes []body into the target interface.
+// It assumes that the response.Result is a SCALE-encoded value.
+func DecodeScale(body []byte, target any) error {
```

Comment:
DecodeScale and Decode are quite similar. There might be a way to merge them or reuse code. For example I changed DecodeScale to below, which seems to pass the test where DecodeScale is getting used.

```
func DecodeScale(body []byte, target any) error {
	var result string
	err := Decode(body, &result)
	if err != nil {
		return err
	}

	resultBytes, err := common.HexToBytes(result)
	if err != nil {
		return fmt.Errorf("convert hex to bytes: %w", err)
	}

	err = scale.Unmarshal(resultBytes, target)
	if err != nil {
		return fmt.Errorf("decode response result: %w", err)
	}

	return nil
}
```